lib/plugins/aws/utils/parse-s3-uri.js

```diff
@@ -1,23 +1,24 @@
 'use strict';
 
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html
 const patterns = [
   // S3 URI. Ex: s3://bucket/path/to/artifact.zip
-  new RegExp('^s3://([^/]+)/(.+)'),
+  new RegExp('^s3://(?<bucket>[^/]+)/(?<key>.+)'),
 
-  // New style S3 URL. Ex: https://bucket.s3.amazonaws.com/path/to/artifact.zip
-  new RegExp('([^/]+)\\.s3\\.amazonaws\\.com/(.+)'),
+  // New style S3 URL. Ex: https://bucket.s3.REGION.amazonaws.com/path/to/artifact.zip
+  new RegExp('(?<bucket>[^/]+)\\.s3(\\.[\\w\\d-]+)?\\.amazonaws\\.com/(?<key>.+)'),
 
-  // Old style S3 URL. Ex: https://s3.amazonaws.com/bucket/path/to/artifact.zip
-  new RegExp('s3\\.amazonaws\\.com/([^/]+)/(.+)'),
+  // Old style S3 URL. Ex: https://s3.REGION.amazonaws.com/bucket/path/to/artifact.zip
+  new RegExp('s3(\\.[\\w\\d-]+)?\\.amazonaws\\.com/(?<bucket>[^/]+)/(?<key>.+)'),
```

Comment:
For this regex and the one above, you should probably support urls with dashes separating the `s3` from the `region` as apparently `s3-region` is a valid url is some aws regions.

Ex: `https://my-bucket.s3-us-west-2.amazonaws.com`

So `s3([.-][\\w\\d-]+)?`.